src/Catalog/PackageCatalogItemCreator.cs

```diff
@@ -198,19 +207,28 @@ private async Task<PackageCatalogItem> GetPackageViaHttpAsync(FeedPackageDetails
             catch (TaskCanceledException tce)
             {
                 // If the HTTP request timed out, a TaskCanceledException will be thrown.
-                throw new HttpClientTimeoutException($"HttpClient request timed out in {nameof(FeedHelpers.DownloadMetadata2CatalogAsync)}.", tce);
+                throw new HttpClientTimeoutException($"HttpClient request timed out in {nameof(PackageCatalogItemCreator.GetPackageViaHttpAsync)}.", tce);
             }
 
             if (response.IsSuccessStatusCode)
             {
                 using (var stream = await response.Content.ReadAsStreamAsync())
                 {
+                    var deprecationItem = packageItem.HasDeprecationInfo
+                        ? new PackageDeprecationItem(
+                            packageItem.DeprecationInfo.Reasons,
+                            packageItem.DeprecationInfo.Message,
+                            packageItem.DeprecationInfo.AlternatePackageId,
+                            packageItem.DeprecationInfo.AlternatePackageRange)
+                        : null;
+
                     item = Utils.CreateCatalogItem(
                         packageItem.ContentUri.ToString(),
                         stream,
                         packageItem.CreatedDate,
                         packageItem.LastEditedDate,
-                        packageItem.PublishedDate);
+                        packageItem.PublishedDate,
+                        deprecationItem: deprecationItem);
```

Comment:
Same as above: can we not just pass `packageItem.DeprecationInfo` here directly? Why do we need to clone the object?

Reply:
No longer needed indeed. Fixed in https://github.com/NuGet/NuGet.Services.Metadata/pull/542/commits/c9193c2f337d98a07658c107296d03b815243964